tests/test_views.py

```diff
@@ -0,0 +1,455 @@
+            password="test_password123"
+        )
+        self.manufacturer = Manufacturer.objects.create(
+            name="Audi",
+            country="Test Country"
+        )
+        self.manufacturer_2 = Manufacturer.objects.create(
+            name="BMW",
+            country="Test Country2"
+        )
+        self.all_manufacturers = Manufacturer.objects.all()
+        self.manufacturer_data = {
+            "name": "Updated Name",
+            "country": "Updated Country"}
+        self.client.force_login(self.user)
+
+    def test_logged_in_user_access(self) -> None:
+        for url in (
+                MANUFACTURER_LIST_URL,
+                MANUFACTURER_CREATE_URL,
+                get_manufacturer_update_url(self.manufacturer),
+                get_manufacturer_delete_url(self.manufacturer)
+        ):
+            http_response = self.client.get(url)
+            self.assertEqual(
+                http_response.status_code,
+                200
+            )
+
+    def test_list(self) -> None:
+        http_response = self.client.get(MANUFACTURER_LIST_URL)
+
+        self.assertEqual(
+            list(http_response.context["manufacturer_list"]),
+            list(self.all_manufacturers)
+        )
+
+    def test_list_search(self) -> None:
+        for search_letter in ("a", "b", "s"):
+            http_response = self.client.get(
+                MANUFACTURER_LIST_URL,
+                {"name": f"{search_letter}"}
+            )
+            self.assertEqual(
+                list(http_response.context["manufacturer_list"]),
+                list(self.all_manufacturers.filter(
+                    name__icontains=f"{search_letter}")
+                )
+            )
+
+    def test_create(self) -> None:
+        http_response = self.client.post(
+            MANUFACTURER_CREATE_URL,
+            data=self.manufacturer_data
+        )
+        manufacturer = Manufacturer.objects.get(
+            name=self.manufacturer_data["name"]
+        )
+
+        self.assertEqual(
+            manufacturer.country,
+            self.manufacturer_data["country"]
+        )
+        self.assertEqual(
+            http_response.status_code,
+            302
+        )
+
+    def test_update(self) -> None:
+        http_response = self.client.post(
+            get_manufacturer_update_url(self.manufacturer),
+            data=self.manufacturer_data
+        )
+        manufacturer = Manufacturer.objects.get(
+            name=self.manufacturer_data["name"]
+        )
+
+        self.assertEqual(
+            manufacturer.country,
+            self.manufacturer_data["country"]
+        )
+        self.assertEqual(
+            http_response.status_code,
+            302
+        )
+
+    def test_delete(self) -> None:
+        self.client.post(get_manufacturer_delete_url(self.manufacturer))
+        self.client.post(get_manufacturer_delete_url(self.manufacturer_2))
+        self.assertFalse(len(self.all_manufacturers))
```

Comment:
In the `test_delete` method of `PrivateManufacturerTest`, consider verifying that the manufacturer objects are actually deleted from the database by checking that they cannot be retrieved anymore, rather than just checking the length of the queryset.